pkg/specgen/generate/container.go

```diff
@@ -48,24 +49,28 @@ func CompleteSpec(ctx context.Context, r *libpod.Runtime, s *specgen.SpecGenerat
 		s.StopSignal = &sig
 	}
 
+	rtc, err := r.GetConfig()
+	if err != nil {
+		return err
+	}
+	// Get Default Environment
+	defaultEnvs, err := envLib.ParseSlice(rtc.Containers.Env)
+	if err != nil {
+		return errors.Wrap(err, "Env fields in containers.conf failed to parse")
+	}
+
 	// Image envs from the image if they don't exist
-	// already
-	env, err := newImage.Env(ctx)
+	// already, overriding the default environments
+	imageEnvs, err := newImage.Env(ctx)
 	if err != nil {
 		return err
 	}
 
-	if len(env) > 0 {
-		envs, err := envLib.ParseSlice(env)
-		if err != nil {
-			return err
-		}
-		for k, v := range envs {
-			if _, exists := s.Env[k]; !exists {
-				s.Env[v] = k
-			}
-		}
+	envs, err := envLib.ParseSlice(imageEnvs)
+	if err != nil {
+		return errors.Wrap(err, "Env fields from image failed to parse")
 	}
+	s.Env = envLib.Join(envLib.Join(defaultEnvs, envs), s.Env)
```

Comment:
this won't respect the priorities, we need to ensure `--env` variables always have a higher priority than what is specified in the image.
My patch could be simplified to not propagate two different Env blocks if the env files coming from `--env-host` has a higher priority than the image's configuration, but I think as an user I'd not expect that.

Reply:
No this does respoend it.  S.Env takes priority, which is input from the users.

defaultEnvs -- containers.conf,
envs == image
s.Env == (--env-host, env-file, --env)

envlib.Join works like append in the second parameter will replace first, if they conflict.

Reply:
BTW if this was not done correctly, I would expect the tests to pass.